rust/kcl-lib/src/execution/state.rs

```diff
@@ -45,33 +44,39 @@ pub(super) struct GlobalState {
     pub id_to_source: IndexMap<ModuleId, ModuleSource>,
     /// Map from module ID to module info.
     pub module_infos: ModuleInfoMap,
+    /// Module loader.
+    pub mod_loader: ModuleLoader,
+    /// Errors and warnings.
+    pub errors: Vec<CompilationError>,
+    #[allow(dead_code)]
```

Comment:
This should only allow dead_code if the feature isn't used, right? In general, I'm not a fan of disabling helpful compiler checks.

Reply:
I'll address in a follow-up